src/index.ts

```diff
@@ -110,37 +80,86 @@ export function lambdaHandler<TContext extends BaseContext>(
         },
         body: httpGraphQLResponse.completeBody,
       };
-    } catch (error) {
-      throw error;
+    } catch (e) {
+      return {
+        statusCode: 400,
+        body: (e as Error).message,
+      };
     }
   };
 }
 
-function createGraphQLRequest(
-  parsedBody: string | object | undefined,
-  event: APIGatewayProxyEvent | APIGatewayProxyEventV2,
-): HTTPGraphQLRequest {
-  const headers = new Map<string, string>();
-  for (const [key, value] of Object.entries(event.headers)) {
-    if (value !== undefined) {
-      // Node/Express headers can be an array or a single value. We join
-      // multi-valued headers with `, ` just like the Fetch API's `Headers`
-      // does. We assume that keys are already lower-cased (as per the Node
-      // docs on IncomingMessage.headers) and so we don't bother to lower-case
-      // them or combine across multiple keys that would lower-case to the
-      // same value.
-      headers.set(key, Array.isArray(value) ? value.join(', ') : value);
-    }
+function normalizeGatewayEvent(event: GatewayEvent): HTTPGraphQLRequest {
+  if (isV1Event(event)) {
+    return normalizeV1Event(event);
+  } else if (isV2Event(event)) {
+    return normalizeV2Event(event);
+  } else {
+    throw Error('Unknown event type');
   }
```

Comment:
To be clear, you're not claiming there's a semantic difference or correction to make here, just a preference? I find if/else if/else to be a more readable construct since it enumerates the possible outcomes clearly like a switch statement with a default. I only slightly prefer it to your suggestion for that reason, but this is a simple enough case that it's readable either way.

Reply:
Correct. Not really a difference. I agree that in this case it doesn't matter but I'll explain why I prefer repeated if's over if-else.

It was a convention used at a previous place I worked, with the reasoning being that on more complex types the if-return would "pick away" from the union. We did this because it would keep consistency between the default cases. We had a lot of statements like the one above, where the default case should not be encountered at runtime and throw, but we also had cases where 2 of 4 possible types of a union would have an early return, and the remaining two would be processed by the default case.

It was the difference between having that default case be indented or not, and the preference was to keep the default case as part of the main function body as opposed to the if-else body.